### PR TITLE
Add linting for parameter defaults

### PR DESCRIPTION
**Pull request by akx**
Refs valohai/valohai-cli#128

### CODE REVIEW

File: valohai_yaml/__main__.py [hunk 1]
```diff
@@ -2,31 +2,43 @@
...
     ap = argparse.ArgumentParser()
+    ap.add_argument(
+        '--strict-warnings',
+        action='store_true',
+        help='exit with error if there are warnings',
+    )
     ap.add_argument('file', nargs='+', help='file(s) to validate')
     args = ap.parse_args(args)
-    errors = 0
+    errors = warnings = 0
```
Comment (magdapoppins) [inline valohai_yaml/__main__.py:17]: Intressant!

File: tests/warning_examples/invalid-numeric-range-parameter-default.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+---
+
+- step:
+    name: test
+    image: tensorflow/tensorflow
+    command: python -u /tmp/code/my_mnist.py
+    parameters:
+      - default:
+        - 1337
```
Comment (magdapoppins) [inline tests/warning_examples/invalid-numeric-range-parameter-default.yaml:9]: Not related to this pr, but how come default is a list when the type is integer? What am I missing, I'd expect it to be an integer.

**Review by magdapoppins — APPROVED** [review]
Looks good! Was there a ticket for this or were you divinely inspired?

**Comment by akx** [comment]
> Looks good! Was there a ticket for this or were you divinely inspired?

https://github.com/valohai/valohai-cli/issues/128 as linked in the description is related, but this doesn't directly fix/close it.

**Pull request merged**